applications/ls/src/lib.rs

```diff
@@ -34,13 +35,21 @@ pub fn main(args: Vec<String>) -> isize {
         return 0;
     }
 
+    let size_option = matches.opt_present("s");
+
     let Ok(curr_wd) = task::with_current_task(|t| t.get_env().lock().working_dir.clone()) else {
         println!("failed to get current task");
         return -1;
     };
+
+    if size_option && matches.free.is_empty() {
+        print_children(&curr_wd, true);
+        return 0;
+    }
+
     // print children of working directory if no child is specified
     if matches.free.is_empty() {
-        print_children(&curr_wd);
+        print_children(&curr_wd, false);
```

Comment:
Suggestion: reduce the above code duplication with just a single call:
```rust
    print_children(&curr_wd, size_option);
```

Reply:
Got it thank you - I will add that change